tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -113,7 +113,7 @@ void start(String[] args) throws IOException {
                 record = new ProducerRecord<>(topicName, payload);
 
                 long sendStartMs = System.currentTimeMillis();
-                Callback cb = stats.nextCompletion(sendStartMs, payload.length, stats);
+                cb = stats.nextCompletion(sendStartMs, payload.length, stats);
```

Comment:
Does it need to pass itself (`stats`) to nextCompletion? It seems to me line#399 could be replaced by `new PerfCallback(this.iteration, start, bytes, this);`

Reply:
A great and careful code optimization.